strr-web/components/bcros/status-card/StatusCard.vue

```diff
@@ -28,22 +28,25 @@
         :label="tRegistrationStatus('view')"
         class-name="px-4 py-1 mobile:grow-0"
         variant="ghost"
+        data-test-id="view-application-link"
       />
 
       <BcrosButtonsPrimary
-        v-if="isCertificateIssued"
-        :action="() => downloadCertificate(registrationId.toString())"
-        :label="tRegistrationStatus('download')"
+        v-if="isPaid"
+        :loading="downloadingReceipts.has(applicationNumber)"
+        :action="() => downloadReceipt(applicationNumber)"
+        :label="downloadingReceipts.has(applicationNumber) ? '' : tRegistrationStatus('downloadReceipt')"
         class-name="px-4 py-1 mobile:grow-0"
         variant="ghost"
+        data-test-id="download-receipt-link"
       />
     </div>
   </div>
 </template>
 
 <script setup lang="ts">
 import { HostApplicationStatusE, ExaminerApplicationStatusE } from '#imports'
-const { downloadCertificate } = useDownloadCertificate()
+const { downloadReceipt, downloadingReceipts } = useDownloadReceipt()
```

Comment:
Just to clarify, the `downloadingReceipts` will temp hold all the receipts to download, and prevents multiple download actions?

Reply:
Prevent multiple of same applicationNumber from downloading when one is in-progress.

Reply:
That does not seem to work, I can click many times and it keeps downloading.
Also, if an application has a registration attached the receipt says: Payment due.
See: 
![image](https://github.com/user-attachments/assets/a793115e-020c-45ca-bd34-31d9f82cdfa9)

Reply:
![image](https://github.com/user-attachments/assets/bd5b852d-3034-4800-8fcb-896bd707f99e)

Reply:
For registry-dashboard, I just checked and it is working as expected. Can you try that on Dev site?
@rstens